Find a functionally equivalent code:com.android.systemui.settings.BrightnessController.updateSlider:COMMENT
Method Modifier: private     
Comment:/**
 * Fetch the brightness from the system settings and update the slider
 */

@@ -1,10 +1,12 @@
 {
-    int value;
-    try {
-        value = Settings.System.getIntForUser(mContext.getContentResolver(), Settings.System.SCREEN_BRIGHTNESS, UserHandle.USER_CURRENT);
-    } catch (SettingNotFoundException ex) {
-        value = mMaximumBacklight;
+    if (mAutomatic) {
+        float value = Settings.System.getFloatForUser(mContext.getContentResolver(), Settings.System.SCREEN_AUTO_BRIGHTNESS_ADJ, 0, UserHandle.USER_CURRENT);
+        mControl.setMax((int) BRIGHTNESS_ADJ_RESOLUTION);
+        mControl.setValue((int) ((value + 1) * BRIGHTNESS_ADJ_RESOLUTION / 2f));
+    } else {
+        int value;
+        value = Settings.System.getIntForUser(mContext.getContentResolver(), Settings.System.SCREEN_BRIGHTNESS, mMaximumBacklight, UserHandle.USER_CURRENT);
+        mControl.setMax(mMaximumBacklight - mMinimumBacklight);
+        mControl.setValue(value - mMinimumBacklight);
     }
-    mControl.setMax(mMaximumBacklight - mMinimumBacklight);
-    mControl.setValue(value - mMinimumBacklight);
 }

